### ISSUE TITLE
[Bug]: Medication X button passes wrong ID to server, unmasking deleteOnCloseRxBox archival bug

### ISSUE THREAD

**Issue by LiamStanziani** [openo-beta/Open-O#2339]
### Description

> **Note:** These bug numbers are based on a google document listing medication issues found, not other issues tracked in GitHub.
                                                                                                                                                                                                                                            
  ## Summary
  Two related bugs in the prescription staging UI. Bug `#5` is currently masking Bug `#6` — fixing `#5` without `#6` causes unintended drug archival. Both must ship together.                                                              
                                                                                                                                                                                                                                            
  ### Bug `#5`: X Button Passes Wrong ID to Server [REGRESSION]                                                                                                                                                                             
                                                                                                                                                                                                                                            
  `deletePrescribingDrugFromUI(uiRefId, drugId)` passes `drugId` (DrugReferenceId) to `deletePrescribe()` instead of `uiRefId` (randomId). The server looks up by randomId, finds nothing, and the stash item is never removed.             
                  
  For brand-new drugs, DrugReferenceId defaults to 0 and randomId is a large random number — they will never match. The X button always fails to remove new drugs from the stash.                                                           
                  
  **File:** `SearchDrug3.jsp` — `deletePrescribingDrugFromUI()      

---                                                                                                                                                                     
                                                                                                                                                                                                                                            
  ### Bug `#6`: deleteOnCloseRxBox Flag Never Reset [PRE-EXISTING] [MASKED BY BUG `#5`]                                                                                                                                                             
                                                                                                                                                                                                                                            
  ▎ Note: This bug is currently masked by Bug `#5`. Because Bug `#5` passes the wrong parameter to deletePrescribe(), the server-side archival lookup fails silently. If Bug `#5` is fixed without also fixing Bug `#6`, the issue described below  
  becomes active.                                                                                                                                                                                                                           
                                                                                                                                                                                                                                            
When "Edit Rx" is clicked, updateDeleteOnCloseRxBox() sets deleteOnCloseRxBox to 'true'. It is never reset to 'false'. After Edit Rx, every subsequent X button click also triggers deleteRxOnCloseRxBox(), which archives the drug from the database without confirmation.
                                                                                                                                                                                                                                            
  File: SearchDrug3.jsp — deleteOnCloseRxBox hidden input, updateDeleteOnCloseRxBox()                                                                                                                                                                                                                                                                                      
                                                                                                                                                                                                                                            
  ### Proposed Fix                                                                                                                                                                                                                              
                                                                                                                                                                                                                                            
  - Bug `#5`: Change this.deletePrescribe(drugId) to this.deletePrescribe(uiRefId) in deletePrescribingDrugFromUI()                                                                                                                           
  - Bug `#6`: Reset deleteOnCloseRxBox to 'false' after the lightwindow closes or after Save & Print completes
                                                                                                                                                                                                                                            
  ---                                                                  

### Steps to Reproduce

### Bug `#5`  

**Steps to reproduce:**                                                                                                                                                                                                                   
  1. Open the patient's medication page
  2. Add any drug (not through ReRx — through the search bar only)
  3. Click the X button on the drug card to remove it                                                                                                                                                                                       
  4. Check stash count via browser console:
     ```js                                                                                                                                                                                                                                  
     new Ajax.Request(ctx + '/oscarRx/WriteScript.do?parameterValue=checkNoStashItem&rand=1', {method:'get', onSuccess:function(t){console.log('stash items: ' + t.responseText.evalJSON().NoStashItem)}})                                                                                                                                                                    

### Bug `#6`

 **Steps to reproduce (with Bug `#5` fix applied):**                                                                                                                                                                                         
  1. Add Drug A and Drug B, click "Save & Print" — both print and are saved to DB
  2. Click "Edit Rx" — this sets deleteOnCloseRxBox = 'true'                                                                                                                                                                                
  3. Click X on Drug B's staging card                                                                                                                                                                    

---

### Expected Result

### Bug `#5`
**Expected:** stash items: 0 — drug removed from UI and server stash        

### Bug `#6`
**Expected:** Drug B removed from staging only (the X button should un-stage, not archive)   

---

### Actual Result

### Bug `#5`
**Actual:** stash items: 1 — drug removed from UI only 

### Bug `#6`
**Actual:** Because deleteOnCloseRxBox is still 'true', deletePrescribe() also calls deleteRxOnCloseRxBox(), which archives Drug B from the DB — the same effect as clicking "Del" in the drug profile, but without a confirmation dialog. Drug B shows crossed out and disappears from the default "Current" medication history view after refreshing

---

### Module/Area

Other

### Error Messages/Logs

```shell
N/A
```

### Version/Branch

develop

### Checklist

- [x] I have searched for similar issues
- [x] I can reproduce this consistently
